### ISSUE TITLE
Viewport size_changed emitted before viewport transform updated on window maximize

### ISSUE THREAD

**Issue by BraindeadBZH** [godotengine/godot#7203]
**Operating system or device - Godot version:**
Happening on version 2.1.1 Steam, Windows 10

**Issue description** (what happened, and what was expected):
I have a custom Node2D where I draw a grid within _draw using draw_line.
To optimize I limit the drawing of the grid to the visible area.
To do so, I use get_viewport_rect() and get_viewport_transform().get_origin() to get the view rect.
To update the grid when the window is being resized I have connected the size_changed signal to the update function.
Unfortunately when I click the maximize window button, get_viewport_rect() gives and updated value but not get_viewport_transform().get_origin().
Strangely, if I resize the window manually, then both values are updated.

**Steps to reproduce:**
1. Create a custom Node2D
2.  Create a _draw function like this:
```gdscript
func _draw():
	var clip = get_viewport_rect()
	var origin = get_viewport_transform().get_origin()
	print("Origin: :", origin)
	var pos = clip.pos - origin
	var end = clip.end - origin
	
	# Vertical lines
	for x in range(pos.x, end.x):
		if(x == 0):
			draw_line(Vector2(x, pos.y), Vector2(x, end.y), Color(1.0, 1.0, 1.0, 0.5))
		elif(x % 40 == 0):
			draw_line(Vector2(x, pos.y), Vector2(x, end.y), Color(1.0, 1.0, 1.0, 0.25))
	
	# Horizontal lines
	for y in range(pos.y, end.y):
		if(y == 0):
			draw_line(Vector2(pos.x, y), Vector2(end.x, y), Color(1.0, 1.0, 1.0, 0.5))
		elif(y % 40 == 0):
			draw_line(Vector2(pos.x, y), Vector2(end.x, y), Color(1.0, 1.0, 1.0, 0.25))
```
3. Maximize the window
4. The grid does not fill the entire window

**Comment by bojidar-bg** [comment]
Does this happen on other OS-s too? Also, would you mind if you make a sample project?

**Comment by Noshyaar** [comment]
First of all thank you for your report and sorry for the delay.

We released Godot 3.0 in January 2018 after 18 months of work, fixing many old issues either directly, or by obsoleting/replacing the features they were referring to.

We still have hundreds of issues whose relevance/reproducibility needs to be checked against the [current stable version](https://godotengine.org/download), and that's where you can help us.
**Could you check if the issue that you described initially is still relevant/reproducible in Godot 3.0 or any newer version, and comment about it here?**

For bug reports, please also make sure that the issue contains detailed steps to reproduce the bug and, if possible, a zipped project that can be used to reproduce it right away. This greatly speeds up debugging and bugfixing tasks for our contributors.

Our Bugsquad will review this issue more in-depth in 15 days, and potentially close it if its relevance could not be confirmed.

Thanks in advance.

*Note: This message is being copy-pasted to many "stale" issues (90+ days without activity). It might happen that it is not meaningful for this specific issue or appears oblivious of the issue's context, if so please comment to notify the Bugsquad about it.*

**Comment by akien-mga** [comment]
As there was no answer to the above request, we're closing this issue as inactive.
If you can still reproduce it, please open a new issue against a currently supported release of Godot.
